### ISSUE TITLE
Add Hosted Apache Solr search integration

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/jeffgeerling-com#54]
I currently use [Hosted Apache Solr](https://hostedapachesolr.com) to host a Drupal 7 search index for my site. I need to create a new index for Drupal 8 so I can start using that with Search API and Search API Solr, and recreate the search functionality on JeffGeerling.com. Note that this issue is a prerequisite to #2.

Details for [subscription ID 1181](https://hostedapachesolr.com/node/1181) from Hosted Apache Solr:

  - Solr host: `solr-1181.hostedapachesolr.com`
  - Solr port: `80`
  - Solr core: `5hE5u_jeffgeerling_com`
  - Basic HTTP authentication: [see password on other monitor...]

**Comment by geerlingguy** [comment]
To get the search excerpts to display, I went to the search index 'Processors' and enabled the 'Highlight' processor (thanks to some advice from this blog post: [How to highlight search results in Search API Solr View in Drupal 8](https://www.valuebound.com/resources/blog/how-highlight-search-results-search-api-solr-view-drupal-8)).

During the live stream, we got titles to appear, but couldn't get the body text to appear:

<img width="579" alt="Screen Shot 2020-04-28 at 11 59 53 AM" src="https://user-images.githubusercontent.com/481677/80515581-c93b2000-8947-11ea-898d-fe5a54b30aae.png">

After enabling the Highlight processor:

<img width="1004" alt="Screen Shot 2020-04-28 at 12 00 26 PM" src="https://user-images.githubusercontent.com/481677/80515637-e374fe00-8947-11ea-81f0-1ca651fe3cf2.png">

Now, on to theming!

**Comment by geerlingguy** [comment]
It seems like the disconnect on the live stream came from config not having a password stored; I checked after exporting config again (and re-saving the password), and it showed the password in plain text, e.g.:

```
@@ -44,7 +44,7 @@ backend_config:
     finalize_timeout: 30
     commit_within: 1000
     username: 'username_here'
-    password: ''
+    password: password_here
     solr_version: ''
     http_method: AUTO
     jmx: false
```

So I opened an issue on Drupal.org [Prevent password from being stored in configuration](https://www.drupal.org/project/search_api_solr/issues/3131823), and we'll see what other people say. I want to keep this codebase open source, but there's no way I can do that if there's a plaintext password that has to be in the exported config so Solr search works!

**Comment by geerlingguy** [comment]
Ha, I'm working on implementing facets for the search page, and I ran into this old gem on my own YouTube channel: https://www.youtube.com/watch?v=J8Rt6HSzrqY

**Comment by geerlingguy** [comment]
Haha, how deep the rabbit hole goes; another article I found was [Faceted Search in Drupal 8: Using Search API Solr and Facets](https://dev.acquia.com/blog/search-strategies-in-drupal-8/faceted-search-in-drupal-8-using-search-api-solr-and-facets/06/05/2016/10406)... and was also by me :P

**Comment by geerlingguy** [comment]
Search is _mostly_ working now, though I'm not sure why Facets is returning some weird data, should have many more articles from 2016 and beyond here...

<img width="972" alt="Screen Shot 2020-04-28 at 1 27 34 PM" src="https://user-images.githubusercontent.com/481677/80523693-14f3c680-8954-11ea-9642-c8027e8baef9.png">

**Comment by geerlingguy** [comment]
I got search working well; a few notes:

  1. I had to select `Body » Processed text` and `Comments » Number of comments`, and I also added boost to the `Title` field by setting it to 'Fulltext'
  2. For the facets, I had to spend about 20 minutes tweaking all the ordering and display settings so they worked in a manner that felt 'usable' to me (e.g. short by year, descending, don't show the option for '0' comments, etc.)

Search API Solr, Facets, and Search API Views are definitely an area of Drupal where it's more 'art' than 'science', though. But in the end, the search results are working slightly better than what they were in Drupal _7_, so I'm happy :)

I also use one exposed form block for both the sitewide footer search form and the search page search form, so theming was very slightly simplified there :)
